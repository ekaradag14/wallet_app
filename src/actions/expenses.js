import { v4 as uuidv4 } from "uuid";
//Aqui desestructamos el primero objeto. Si tenemos ninguna tenemos un valor defecto. (Un objeto vacio)
export const addExpense = ({
  description = "",
  note = "",
  amount = 0,
  createdAt = 0
} = {}) => ({
  type: "ADD_EXPENSE",
  expense: {
    id: uuidv4(),
    description,
    note,
    amount,
    createdAt
  }
});
//Estas funciones son solo algunos atajos para no escribir tipo cada vez escribimos las funciones
export const removeExpense = ({ id = "" } = {}) => ({
  type: "REMOVE_EXPENSE",
  id
});
export const editExpense = (id, updates) => ({
  type: "EDIT_EXPENSE",
  id,
  updates
});
