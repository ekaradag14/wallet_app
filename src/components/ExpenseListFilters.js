import "react-dates/lib/css/_datepicker.css";
import React, { useState } from "react";
import {
  setTextFilter,
  sortByDate,
  setStartDate,
  setEndDate,
  sortByAmount
} from "../actions/filters";
import { connect } from "react-redux";
import { DayPickerRangeController } from "react-dates";
import moment from "moment";
const ExpenseListFilters = ({
  putTextFilter,
  putSortDateFilter,
  putAmountFilter,
  putStartDateFilter,
  putEndDateFilter,
  filterValue
}) => {
  const [expenseFilterController, setExpenseFilterController] = useState({
    isInputNecessary: false,
    isCalendarNecessary: false,
    focusedInput: "startDate",
    startDate: null,
    endDate: null
  });
  // Para controlar el estado complejo
  const handleChange = (input) => {
    const { value, name } = input;

    setExpenseFilterController((prev) => {
      return {
        ...prev,
        [name]: value,
      };
    });
  };

  const resetFilters = () => {
    //Antes de ordenar nuestra lista. Necesitamos limpiar nuestra filtros.
    //Tomamos nuestra lista directamente despuÃ©s de filtrarlo
    putStartDateFilter(null);
    putEndDateFilter(null);
    putTextFilter("");
  };

  const handleExtraComponents = (calendarComponent, textField) => {
    //Solo necesitamos estas zonas si usamoslos
    handleChange({ name: "isCalendarNecessary", value: calendarComponent });
    handleChange({ name: "isInputNecessary", value: textField });
  };
  const handleSetFilter = (filterType) => {
    switch (filterType) {
      case "Sort By Date":
        handleExtraComponents(false, false);
        resetFilters();
        //Y llamamos nuestra mascara para dispatch(sortByDate())
        putSortDateFilter();
        break;
      case "Sort By Amount":
        handleExtraComponents(false, false);
        resetFilters();
        putAmountFilter();
        break;
      case "Put Text Filter":
        resetFilters();
        handleExtraComponents(false, true);
        break;
      case "Put Date Filter":
        resetFilters();
        handleExtraComponents(true, false);
        break;
      default:
    }
  };

  //Necesario para react componente de airbnb
  const onFocusChange = (focusedInput) => {
    handleChange({
      name: "focusedInput",
      value: !focusedInput ? "startDate" : focusedInput
    });
  };

  //Necesario para react componente de airbnb
  const onDatesChange = ({ startDate, endDate }) => {
    handleChange({ name: "startDate", value: startDate });
    handleChange({ name: "endDate", value: endDate });
    startDate && putStartDateFilter(startDate.valueOf());
    endDate && putEndDateFilter(endDate.valueOf());
  };

  return (
    <div>
      <select
        //Tiene los atributos mismos de input
        value={filterValue}
        onChange={(e) => handleSetFilter(e.target.value)}
      >
        <option>Sort By Date</option>
        <option>Sort By Amount</option>
        <option>Put Text Filter</option>
        <option>Put Date Filter</option>
      </select>

      {expenseFilterController.isInputNecessary && (
        <input
          type="text"
          value={filterValue}
          onChange={(e) => putTextFilter(e.target.value)}
        />
      )}
      {/* Puedes encontrar las docs sobre este componente en github o site de airbnb*/}
      {expenseFilterController.isCalendarNecessary && (
        <DayPickerRangeController
          startDate={expenseFilterController.startDate}
          endDate={expenseFilterController.endDate}
          onDatesChange={onDatesChange}
          focusedInput={expenseFilterController.focusedInput}
          onFocusChange={onFocusChange}
          initialVisibleMonth={() => moment()}
      
        />
      )}
    </div>
  );
};

//Mapeamos algunos atributos para utilizar funciones directamente.
//react-redux toma dispatch directamente
const mapDispatchToProps = (dispatch) => {
  return {
    //Estas funciones son solamente algunos atajos para usar mas bien que escribir el objeto directamente
    putTextFilter: (filter) => dispatch(setTextFilter(filter)),
    putStartDateFilter: (date) => dispatch(setStartDate(date)),
    putEndDateFilter: (date) => dispatch(setEndDate(date)),
    putSortDateFilter: () => dispatch(sortByDate()),
    putAmountFilter: () => dispatch(sortByAmount())
  };
};

//Usamos valor de filtro en input
const mapsStateToProps = (state) => {
  return {
    filterValue: state.filter
  };
};
//El primer para obtener el estado de store, y segundo para obtener acciones. y Por ultimo envolvemos nuestro componente
export default connect(
  mapsStateToProps,
  mapDispatchToProps
)(ExpenseListFilters);
