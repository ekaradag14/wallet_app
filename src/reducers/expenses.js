//Si tenemos un valor defecto complejo estas valores defectos sera util
const expensesReducerDefaultState = [];
//Cambiamos estado del expenses en redux con funciones de acciones
const expensesReducer = (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_EXPENSE':
      //AÃ±adimos un expense nuevo
      return [...state, action.expense];
    case 'REMOVE_EXPENSE':
      return state.filter((expense) => expense.id !== action.id);
    case 'EDIT_EXPENSE':
      return state.map((expense) => {
        //Actualizamos un objeto soltero que concuerda con nuestra id
        if (expense.id === action.id) {
          return {
            //Todas las actualizaciones sobrescriben el estado previo
            ...expense,
            ...action.updates,
          };
        } else {
          return expense;
        }
      });

    default:
      return state;
  }
};

export default expensesReducer;
