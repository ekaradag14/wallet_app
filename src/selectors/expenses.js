const getVisibleExpenses = (expenses, { text, sortBy, startDate, endDate }) => {
  return expenses
    .filter((expense) => {
      //Usamos || por utilizar filtros aunque no tenemos todos los valores para filtros
      const startDateMatch =
        typeof startDate !== 'number' || expense.createdAt >= startDate;
      const endDateMatch =
        typeof endDate !== 'number' || expense.createdAt <= endDate;
      const textMatch =
        typeof text !== 'string' ||
        expense.description.toLowerCase().includes(text.toLocaleLowerCase());
      //Devolvemos un objeto si concuerdas a todos los filtros
      return startDateMatch && endDateMatch && textMatch;
    })
    .sort((a, b) => {
      //sort funciÃ³n de javascript.
      if (sortBy === 'date') {
        // Ordenamos de nuevo a viejo
        return a.createdAt > b.createdAt ? 1 : -1; // En sort si queremos a ven antes de b devolvemos -1
      } else if (sortBy === 'amount') {
        return a.amount < b.amount ? 1 : -1;
      }
      return 0;
    });
};

export default getVisibleExpenses;
