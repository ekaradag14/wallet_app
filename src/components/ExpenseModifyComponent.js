import "react-dates/lib/css/_datepicker.css";
import React, { useState } from "react";
import { connect } from "react-redux";
import { addExpense, editExpense } from "../actions/expenses";
import { SingleDatePicker } from "react-dates";
import moment from "moment";


export const ExpenseModifyComponent = (props) => {
  const [expenseName, setExpenseName] = useState("");
  const [expenseNote, setExpenseNote] = useState("");
  const [expenseValue, setExpenseValue] = useState("");
  const [expenseDate, setExpenseDate] = useState("");
  const [expenseError, setExpenseError] = useState('');
  const [isCalendarFocused, setIsCalendarFocused] = useState(false);

  //Si tenemos un expense(Estamos editando) pasamoslo
  React.useEffect(() => { //Usamos con react porque necesitamos el para usar in jest. (No hay ninguna lÃ³gica es solo funciona asi)
    if (!!props.expense) {
      setExpenseName(props.expense.description);
      setExpenseValue(props.expense.amount);
      setExpenseNote(props.expense.note);
      setExpenseDate(moment(props.expense.createdAt));
    }
  }, []);

  const onAmountChange = (e) => {
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      //REGEX Code for 234.34 like numbers
      setExpenseValue(amount);
    }
  };

  //Hicimos como asi porque dicen lo en docs de airbnb
  const onFocusChange = ({ focused }) => {
    setIsCalendarFocused(focused);
  };

  const onDateChange = (date) => {
    //Si hay un dato el usuario no tiene eliminar ninguna letra
    if (date) {
      setExpenseDate(date);
    }
  };
  // Fin de airbnb docs

  const handleModifyExpense = (e) => {

    e.preventDefault();
    //Si hay un expense objeto en props (Cuando editamos)
    if (!!props.expense) {
      //Pasamos nuevo expense a nuestra reducer en redux
      props.dispatch(
        editExpense(props.expense.id, {
          description: expenseName,
          note: expenseNote,
          amount: expenseValue,
          createdAt: expenseDate.valueOf()
        })
      );
      props.processCompleted();
    } else if (!props.expense) {
      //Nuevo expense
      
      if (!expenseName || !expenseValue) {
        // Son esencial
        setExpenseError('Please provide a valid description and amount.');
      } else {
        props.dispatch(
          addExpense({
            description: expenseName,
            note: expenseNote,
            amount: expenseValue,
            createdAt: expenseDate.valueOf()
          })
        );
        setExpenseName("");
        setExpenseValue("");
        // A la componente de tablero
        props.processCompleted();
      }
    } else {
      console.log("no match for the button");
    }
  };
  return (
    <div>
      {expenseError && <p className='expenses-modify-component-errors' >{expenseError}</p>}
      {props.expense ? <h3> Edit Expense </h3> : <h3> Add New Expense </h3>}
      <SingleDatePicker
        date={expenseDate || moment()} //  or null
        onDateChange={onDateChange}
        focused={isCalendarFocused} // PropTypes.bool
        onFocusChange={onFocusChange} // PropTypes.func.isRequired
        id="your_unique_id" // PropTypes.string.isRequired,
        numberOfMonths={1} //Numero de meses los usarios veran en componente
        isOutsideRange={() => false} //Ahora podemos elegir los dias del pasado
      />
      <form onSubmit={handleModifyExpense}>
        Expense Description:
        <br />
        <input
          onChange={(e) => setExpenseName(e.target.value)}
          value={expenseName}
          type="text"
          autoFocus
          placeholder="Description"
          className='expenses-modify-component-name-input'
        />
        <br />
        Expense Note:
        <br />
        <textarea
          onChange={(e) => setExpenseNote(e.target.value)}
          value={expenseNote}
          type="text"
          autoFocus
          placeholder="Additional Notes"
        />
        <br />
        Expense Amount:
        <br />
        <input
          onChange={(e) => onAmountChange(e)}
          value={expenseValue}
          placeholder="Amount"
          type="number"
        />
        <br />
        <input type='submit' className="add-todo" value={props.expense ? "Edit Todo" : "Add Todo"} />
      </form>
    </div>
  );
};

// El primer argumento es null porque no necesitamos alcanzar al store aqui.
export default connect(null, null)(ExpenseModifyComponent);
