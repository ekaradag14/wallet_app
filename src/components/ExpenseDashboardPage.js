import React from "react";
//Usamos react-redux para alcanzar al store directamente de los componentes
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
//Importamos el función para filtrar nuestra store
import selectExpenses from "../selectors/expenses";
import ExpenseListFilters from "./ExpenseListFilters";

export const ExpenseDashboardPage = ({ expenses }) => { //Exportamos este para probar con jest.js
  const listIt = expenses.map((element) => (
    //Doy todo objeto en un manera rested
    <ExpenseListItem key={element.id} {...element} />
  ));

  return (
    <div>
      This is my expense list
      <ExpenseListFilters />
      {expenses.length === 0 ? <p>I have no expenses :)</p> : listIt }
    </div>
  );
};

const mapsStateToProps = (state) => {
  //Volvemos que necesitamos de store
  return {
    //Solo damos un estado filtrado para mostrar.
    expenses: selectExpenses(state.expenses, state.filters)
  };
  //Ahora tenemos gastos y filtros en props de ExpenseList
};

// El argumento primer es para función que desestructura el store. El argumento secundario es para a cual componente pasamos los partes de store.

export default connect(mapsStateToProps)(ExpenseDashboardPage);
