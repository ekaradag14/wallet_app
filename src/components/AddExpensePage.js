import React from "react";
import ExpenseModifyComponent from "./ExpenseModifyComponent";
export const AddExpensePage = (props) => {
  return (
    <div>
      <ExpenseModifyComponent
        processCompleted={() => {
          //Una vez acabamos modificando de expense redireccionamos el usuario a componente de tablero
          props.history.push("/");
        }}
      />
    </div>
  );
};

// El primer argumento es null porque no necesitamos alcanzar al store aqui.
export default AddExpensePage;
