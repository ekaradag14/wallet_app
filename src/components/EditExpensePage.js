import React from "react";
import ExpenseModifyComponent from "./ExpenseModifyComponent";
import { connect } from "react-redux";

const EditExpensePage = ({ expenses, match, expense,history }) => {
  return (
    <div>
      {expense && <h3>Editing {expense.description} expense </h3>}
      <ExpenseModifyComponent
        expense={expense}
        processCompleted={() => {
          history.push("/");
        }}
      />
    </div>
  );
};

const mapsStateToProps = (state, props) => { // Pasamos el estado para buscar por el expense 
  //Y Después pasamos ese expense a props para usarlo
  return {
    expense: state.expenses.find((item) => item.id === props.match.params.id)
      //match.params.id es id de nuestra expense la url de pagina es dinámica y construimos con la id
  };
};

export default connect(mapsStateToProps)(EditExpensePage);
