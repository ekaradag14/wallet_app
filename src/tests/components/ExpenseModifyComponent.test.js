import React from 'react';
import { shallow } from 'enzyme'
import moment from 'moment'
import {ExpenseModifyComponent} from '../../components/ExpenseModifyComponent'
import toJSON from 'enzyme-to-json'
import expenses from '../fixtures/expenses'
import {SingleDatePicker} from 'react-dates'

describe.skip('Expense Modify Component Test Scenarios', () => {
    let useEffect;
    let mockUseEffect = () => { //Es de una solución de internet
        useEffect.mockImplementationOnce((f) => f());
    };

    test('should render the component with default values', ()=> {
        const wrapper = shallow(<ExpenseModifyComponent/>);
        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    test('should render the component with given values', () => {
        useEffect = jest.spyOn(React, 'useEffect');
        mockUseEffect()
        const wrapper = shallow (<ExpenseModifyComponent expense= {expenses[0]} />)
        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    test('should render error for invalid form submission', () => {
        const wrapper = shallow(<ExpenseModifyComponent  processCompleted={() => {}}/>); //processCompleted es un prop de ExpenseModifyComponent. No damos nada mas props porque queremos que probar un mensaje de error.
        wrapper.find('form').simulate('submit', { preventDefault: () => {},}); //Buscamos un niño de nuestro componente con clase de .add-todo (que es un botón). Pues simulamos del evento de click y nos burlamos preventDefault.
        //Con componentes funcionales no podemos usar estado de wrapper(Podemos usar este con componentes de tipo clase). Por eso vigilamos componente niño donde mostramos mensaje de error.
        expect(wrapper.find(".expenses-modify-component-errors").length).toBeGreaterThan(0);
    })

    test('should control changed input in name input', () => {
        const value = 'Brand New Description';
        const wrapper = shallow(<ExpenseModifyComponent  />); 
        wrapper.find('input[placeholder~="Description"]').simulate('change',{ target : { value }}) 
        // Recuerda: Este tipo de css selector no funcionan en Safari
        expect(wrapper.find('input[placeholder~="Description"]').props().value).toEqual(value); //Controlamos value de input desde no podemos accesar el estado 
    })

    test('should control changed input in description input', () => {
        const value = 'Brand New Note';
        const wrapper = shallow(<ExpenseModifyComponent  />); 
        wrapper.find('textarea').at(0).simulate('change',{ target : { value }}) //at(0) significa buscar el primer textarea en nuestra wrapper
        expect(wrapper.find('textarea').at(0).props().value).toEqual(value); //Controlamos value de input desde no podemos accesar el estado 
    })

    test('should set amount if valid input', () => {
        const value = '23.45';
        const wrapper = shallow(<ExpenseModifyComponent  />); 
        wrapper.find('input[placeholder~="Amount"]').simulate('change',{ target : { value }}) 
        // Recuerda: Este tipo de css selector no funcionan en Safari
        expect(wrapper.find('input[placeholder~="Amount"]').props().value).toEqual(value); //Controlamos value de input desde no podemos accesar el estado 
    })

    test('should set amount if in valid input', () => {
        const value = '23.45';
        const wrongValue = '23.452343';
        const wrapper = shallow(<ExpenseModifyComponent  />); 
        wrapper.find('input[placeholder~="Amount"]').simulate('change',{ target : { value }}) 
        wrapper.find('input[placeholder~="Amount"]').simulate('change',{ target : { value: wrongValue }}) 
        expect(wrapper.find('input[placeholder~="Amount"]').props().value).toEqual(value); //Controlamos value de input desde no podemos accesar el estado 
    })

    test('should call onSubmit prop fro valid form submission',() => {
      const onSubmitSpy = jest.fn();//Burlaremos de un función
      useEffect = jest.spyOn(React, 'useEffect');
      mockUseEffect(); //Necesitamos burlarse de useEffect para poner datos de expense
      const wrapper = shallow(<ExpenseModifyComponent expense={expenses[0]} dispatch={onSubmitSpy}  processCompleted={() => {} }/>);//Burlamos de dispatch
 
      wrapper.find('form').simulate('submit', { preventDefault: () => {},}); //Burlamos de submit del form
      expect(onSubmitSpy).toHaveBeenLastCalledWith({ //La input después de addExpense de acciones 
      "id": 0,
      "type": "EDIT_EXPENSE",
      "updates": {
          "amount": 220,
          "createdAt": 100,
          "description": "Rent",
          "note": "This month's rent",
       }
      })
    })

    test('should set date if date given to SingleDatePicker', () => {
      const wrapper = shallow(<ExpenseModifyComponent />);
      wrapper.find(SingleDatePicker).prop('onDateChange')(moment(1)); //Buscamos la componente SingleDatePicker y damos un prop y llamamos lo
      expect(wrapper.find(SingleDatePicker).prop('date').valueOf()).toBe(moment(1).valueOf())
    })

    test('should set focus on SingleDatePicker', () => {
        
      const wrapper = shallow(<ExpenseModifyComponent />);
      wrapper.find(SingleDatePicker).prop('onFocusChange')({ focused: true}); //Buscamos la componente SingleDatePicker y damos un prop y llamamos lo
      expect(wrapper.find(SingleDatePicker).prop('focused').valueOf()).toBe(true)
    })
})