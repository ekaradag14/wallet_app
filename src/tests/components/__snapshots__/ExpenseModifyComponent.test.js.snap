// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense Modify Component Test Scenarios should render the component with default values 1`] = `
<div>
  <h3>
     Add New Expense 
  </h3>
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    appendToBody={false}
    block={false}
    calendarInfoPosition="bottom"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.000Z"}
    dayPickerNavigationInlineStyles={null}
    daySize={39}
    disableScroll={false}
    disabled={false}
    displayFormat={[Function]}
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    horizontalMonthPadding={13}
    id="your_unique_id"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepFocusOnInput={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPosition="navPositionTop"
    navPrev={null}
    noBorder={false}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsSelected": [Function],
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month.",
        "jumpToPrevMonth": "Move backward to switch to the previous month.",
        "keyboardBackwardNavigationInstructions": "Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardForwardNavigationInstructions": "Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
        "moveFocusByOneMonth": "Switch months.",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
        "openThisPanel": "Open this panel.",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field.",
        "roleDescription": "datepicker",
        "selectFocusedDate": "Select the date in focus.",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    regular={false}
    renderCalendarInfo={null}
    renderDayContents={null}
    renderMonthElement={null}
    renderMonthText={null}
    renderNavNextButton={null}
    renderNavPrevButton={null}
    renderWeekHeaderElement={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    small={false}
    verticalHeight={null}
    verticalSpacing={22}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <form
    onSubmit={[Function]}
  >
    Expense Description:
    <br />
    <input
      autoFocus={true}
      className="expenses-modify-component-name-input"
      onChange={[Function]}
      placeholder="Description"
      type="text"
      value=""
    />
    <br />
    Expense Note:
    <br />
    <textarea
      autoFocus={true}
      onChange={[Function]}
      placeholder="Additional Notes"
      type="text"
      value=""
    />
    <br />
    Expense Amount:
    <br />
    <input
      onChange={[Function]}
      placeholder="Amount"
      type="number"
      value=""
    />
    <br />
    <input
      className="add-todo"
      type="submit"
      value="Add Todo"
    />
  </form>
</div>
`;

exports[`Expense Modify Component Test Scenarios should render the component with given values 1`] = `
<div>
  <h3>
     Edit Expense 
  </h3>
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    appendToBody={false}
    block={false}
    calendarInfoPosition="bottom"
    customCloseIcon={null}
    customInputIcon={null}
    date={"1970-01-01T00:00:00.100Z"}
    dayPickerNavigationInlineStyles={null}
    daySize={39}
    disableScroll={false}
    disabled={false}
    displayFormat={[Function]}
    enableOutsideDays={false}
    firstDayOfWeek={null}
    focused={false}
    hideKeyboardShortcutsPanel={false}
    horizontalMargin={0}
    horizontalMonthPadding={13}
    id="your_unique_id"
    initialVisibleMonth={null}
    inputIconPosition="before"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepFocusOnInput={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={null}
    navPosition="navPositionTop"
    navPrev={null}
    noBorder={false}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsSelected": [Function],
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month.",
        "jumpToPrevMonth": "Move backward to switch to the previous month.",
        "keyboardBackwardNavigationInstructions": "Navigate backward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardForwardNavigationInstructions": "Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
        "moveFocusByOneMonth": "Switch months.",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
        "openThisPanel": "Open this panel.",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field.",
        "roleDescription": "datepicker",
        "selectFocusedDate": "Select the date in focus.",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder="Date"
    readOnly={false}
    regular={false}
    renderCalendarInfo={null}
    renderDayContents={null}
    renderMonthElement={null}
    renderMonthText={null}
    renderNavNextButton={null}
    renderNavPrevButton={null}
    renderWeekHeaderElement={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    small={false}
    verticalHeight={null}
    verticalSpacing={22}
    weekDayFormat="dd"
    withFullScreenPortal={false}
    withPortal={false}
  />
  <form
    onSubmit={[Function]}
  >
    Expense Description:
    <br />
    <input
      autoFocus={true}
      className="expenses-modify-component-name-input"
      onChange={[Function]}
      placeholder="Description"
      type="text"
      value="Rent"
    />
    <br />
    Expense Note:
    <br />
    <textarea
      autoFocus={true}
      onChange={[Function]}
      placeholder="Additional Notes"
      type="text"
      value="This month's rent"
    />
    <br />
    Expense Amount:
    <br />
    <input
      onChange={[Function]}
      placeholder="Amount"
      type="number"
      value={220}
    />
    <br />
    <input
      className="add-todo"
      type="submit"
      value="Edit Todo"
    />
  </form>
</div>
`;
